be/src/vec/common/schema_util.cpp

```diff
@@ -152,13 +152,31 @@ Status cast_column(const ColumnWithTypeAndName& arg, const DataTypePtr& type, Co
     Block tmp_block {arguments};
     size_t result_column = tmp_block.columns();
     auto ctx = FunctionContext::create_context(nullptr, {}, {});
+
+    // To prevent from null info lost, we should not call function since the function framework will wrap
+    // nullable to Variant instead of the root of Variant
+    // Nullable(Array(int)) -> Variant<Nullable(Array(int))>
```

Comment:
Is this example right or wrong? Can you write both a right and a wrong example in comment?

Reply:
right:  Nullable(Array(int)) -> Nullable(Variant<Nullable(Array(int)))
wrong:   Nullable(Array(int)) -> NullableVariant<(Array(int)))

Reply:
done